repo2docker/buildpacks/pipfile/__init__.py

```diff
@@ -65,51 +68,72 @@ def python_version(self):
     def get_assemble_scripts(self):
         """Return series of build-steps specific to this repository.
         """
-        # If we have a runtime.txt & that's set to python-2.7,
-        # requirements.txt will be installed in the *kernel* env
-        # and requirements3.txt (if it exists)
-        # will be installed in the python 3 notebook server env.
+        # If we have either Pipfile.lock, Pipfile, or runtime.txt declare the
+        # use of Python 2, Python 2.7 will be made available in the *kernel*
+        # environment. The notebook servers environment on the other hand
+        # requires Python 3 but may require something additional installed in it
+        # still such as `nbgitpuller`. For this purpose, a "requirements3.txt"
+        # file will be used to install dependencies for the notebook servers
+        # environment, if Python 2 had been specified for the kernel
+        # environment.
         assemble_scripts = super().get_assemble_scripts()
 
         if self.py2:
-            # using python 2 kernel,
-            # requirements3.txt allows installation in the notebook server env
+            # using Python 2 as a kernel, but Python 3 for the notebook server
+
+            # requirements3.txt allows for packages to be installed to the
+            # notebook servers Python environment
             nb_requirements_file = self.binder_path("requirements3.txt")
             if os.path.exists(nb_requirements_file):
                 assemble_scripts.append(
                     (
                         "${NB_USER}",
-                        # want the $NB_PYHTON_PREFIX environment variable, not for
-                        # Python's string formatting to try and replace this
-                        '${{NB_PYTHON_PREFIX}}/bin/pip install --no-cache-dir -r "{}"'.format(
+                        'pip install --no-cache-dir -r "{}"'.format(
```

Comment:
Why not keep this as `${{NB_PYTHON_PREFIX}}/bin/pip` which is what the [Python buildpack](https://github.com/jupyter/repo2docker/blob/cd809645a9e00cf14abd110d8fb8098193379a7c/repo2docker/buildpacks/python/__init__.py#L56-L61) uses?

Reply:
Sure - let's remain explicit!